### DISCUSSION TITLE
grouping leds and controlling groups

### DISCUSSION THREAD

**Discussion by dugly66** [julianschill/klipper-led_effect#231] [Q&A]
Ok I cant figure out how or if I can do this.
I have a total of 58 neopixel leds. 
I want to group them like in mainsail. 
left 1-19
top 20-38
right 39-58
I would like the left to do extruder temp.
right to do bed temp but reversed.
and top to do something else.
can I do it and if so how?

**Comment by julianschill** [comment]
For the effects, define the leds like this
```leds: neopixel:my_neopixel (1-19)```

If you want to group them, look into https://github.com/MapleLeafMakers/klipper-virtual_leds

Make sure to put the grouping before the effects in the config file.

The pull request to Klipper for grouping leds has not yet been accepted yet.

**Reply by dugly66** [reply]
so I can only setup 1 group like that without using virtual leds or can I do 
leds: neopixel:left (1-19)
leds: neopixel:top (20-38)
leds: neopixel:right (58-39)

**Reply by julianschill** [reply]
Like this:
```
leds: neopixel:left (1-19)
      neopixel:top (20-38)
      neopixel:right (58-39)
 ```

**Reply by julianschill** [reply]
But for what you want to do just make three effects and assign them to the respective leds.
```
[neopixel my_chain]
pin: ...
...

[led_effect extruder_temp]
leds: neopixel:my_chain (1-19)
layers:...
...

[led_effect bed_temp]
leds: neopixel:my_chain (58-39)
layers:...
...

[led_effect something_else]
leds: neopixel:my_chain (20-38)
layers:...
...
```

**Reply by dugly66** [reply]
OK well The one thing I have relized now is that your smarter then me when it comes to this lol. 
I just cant wrap my head sround this.
I can call an effect from gcode but for the life of me I can not figure this out.
Thanks for trying to explain it to my smooth brain but I think I will have to give in for now. Maybe come back at it in a week.
